lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1710,6 +1710,54 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     }
     gtTensor.emitError("unimplemented: dtype isn't supported.");
   }
+  if (auto eqTensor = dyn_cast<AtenEqTensorOp>(op)) {
+    AtenEqTensorOp::Adaptor adaptor(operands);
+    Type lhsDtype = payloadArgs[0].getType();
+    Type rhsDtype = payloadArgs[1].getType();
+
+    // TODO: Type promotion in case of different `lhsDtype` and `rhsDtype` needs
+    // to be handled.
+    if (lhsDtype != rhsDtype)
+      eqTensor.emitError("unimplemented: lhs and rhs dtype must be same");
```

Comment:
After emitting an error, make sure you return `nullptr` to avoid executing the rest of the function. The same goes for the other `emitError`s below. Also, I think `gtTensor` above forgets to do this too. It would be nice if you could add return statements to those too